paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/action/CloneActionITCase.java

```diff
@@ -0,0 +1,400 @@
+                        "+I[1, 100, 15, 20221201]",
+                        "+I[12, 312, 90, 20221208]",
+                        "+I[2, 40000, 16, 20221208]",
+                        "+I[24, 898, 1, 20221209]",
+                        "+I[29, 434, 15, 20221209]",
+                        "+I[4, 100, 19, 20221209]");
+        validateResult(table, ROW_TYPE, sourceTableScan, scanResult, 60_000);
+        validateResult(targetTable, ROW_TYPE, targetTableScan, scanResult, 60_000);
+
+        /**
+         * check file name and file size of manifest && index && data && schema && changelog &&
+         * snapshot files
+         */
+        List<Path> targetTableFiles = PickFilesUtil.getUsedFilesForLatestSnapshot(targetTable);
+        List<Pair<Path, Path>> filesPathInfoList =
+                targetTableFiles.stream()
+                        .map(
+                                absolutePath ->
+                                        Pair.of(
+                                                absolutePath,
+                                                getPathExcludeTableRoot(
+                                                        absolutePath, targetTable.location())))
+                        .collect(Collectors.toList());
+
+        Path tableLocation = table.location();
+        for (Pair<Path, Path> filesPathInfo : filesPathInfoList) {
+            Path sourceTableFile = new Path(tableLocation.toString() + filesPathInfo.getRight());
+            Assertions.assertTrue(table.fileIO().exists(sourceTableFile));
+            Assertions.assertEquals(
+                    table.fileIO().getFileSize(sourceTableFile),
+                    targetTable.fileIO().getFileSize(filesPathInfo.getLeft()));
+        }
+    }
+
+    @Test
+    @Timeout(60_000)
+    public void testSnapshotWithSlowExpire() throws Exception {
+        /** prepare source table */
+        prepareTable(
+                Arrays.asList("dt", "hh"),
+                Arrays.asList("dt", "hh", "k"),
+                new HashMap<String, String>() {
+                    {
+                        put(CoreOptions.SNAPSHOT_NUM_RETAINED_MIN.key(), "3");
+                        put(CoreOptions.SNAPSHOT_NUM_RETAINED_MAX.key(), "3");
+                    }
+                });
+
+        /** write data into source table and commit them */
+        Thread writeDataToSourceTablethread =
+                new Thread(
+                        () -> {
+                            try {
+                                /** write and commit data to source table. */
+                                while (true) {
+                                    writeData(
+                                            rowData(
+                                                    1,
+                                                    100,
+                                                    15,
+                                                    BinaryString.fromString("20221208")),
```

Comment:
Too many indention. Extract a method.

Reply:
OK, these code is not duplicated after I delete testSnapshotWithSlowExpire case.